### ISSUE TITLE
[Feature Request]: Can ragflow add a speech-to-text model such as faster whisper? Can only fish audio be added to the text-to-speech model

### ISSUE THREAD

**Issue by ZxnSnowy** [infiniflow/ragflow#2879]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Describe the feature you'd like

I wish I could add speech-to-text models like whisper. There are also text-to-speech models, such as cosyvoice

**Comment by KevinHuSh** [comment]
Do you mean this?
![image](https://github.com/user-attachments/assets/02c4f3fa-4f74-41ba-aa51-d49ee6d30aed)

**Comment by ZxnSnowy** [comment]
> 你是这个意思吗？ ![图像](https://private-user-images.githubusercontent.com/12318111/377690277-02c4f3fa-4f74-41ba-aa51-d49ee6d30aed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0NzUwMzgsIm5iZiI6MTcyOTQ3NDczOCwicGF0aCI6Ii8xMjMxODExMS8zNzc2OTAyNzctMDJjNGYzZmEtNGY3NC00MWJhLWFhNTEtZDQ5ZWU2ZDMwYWVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDAxMzg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MWEyMjYzZmMyNzVkNjljMzE0N2UxZjYxOTg0ODRiMzk5YzFhMjczYzk4ZGZjZTdlZmNhZWJmZTQ2OTA2YjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4y9cfN0n6odmk29ek3_DjosTBF-zZhkqfBnsKFmFbmc)

Yes, how did you achieve that?

**Comment by ZxnSnowy** [comment]
> 你是这个意思吗？ ![图像](https://private-user-images.githubusercontent.com/12318111/377690277-02c4f3fa-4f74-41ba-aa51-d49ee6d30aed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0NzUwMzgsIm5iZiI6MTcyOTQ3NDczOCwicGF0aCI6Ii8xMjMxODExMS8zNzc2OTAyNzctMDJjNGYzZmEtNGY3NC00MWJhLWFhNTEtZDQ5ZWU2ZDMwYWVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDAxMzg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MWEyMjYzZmMyNzVkNjljMzE0N2UxZjYxOTg0ODRiMzk5YzFhMjczYzk4ZGZjZTdlZmNhZWJmZTQ2OTA2YjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4y9cfN0n6odmk29ek3_DjosTBF-zZhkqfBnsKFmFbmc)

Yes, how did you add it?

**Comment by KevinHuSh** [comment]
Add a validated openAI api key.

**Comment by ZxnSnowy** [comment]
> Add a validated openAI api key.

Thank you for your reply. I was wondering if I could add the local model I downloaded? Examples include faster-whisper and GPT-SoVITS.

**Comment by KevinHuSh** [comment]
What about using Xinference to deploy your LLM model？

**Comment by zzc-ccccc** [comment]
> What about using Xinference to deploy your LLM model？

Good idea. I have already used Xinference to deploy CosyVoice2-0.5B. My question is: How can we use this model. Do we need to develop the audio workflow and get the text data back from xinderence service?

**Comment by Yannnnnnny** [comment]
<img width="1251" height="524" alt="Image" src="https://github.com/user-attachments/assets/3125284e-5eb3-4dfd-a840-e4b11fa2eb07" />
Since the issue has been resolved, I'll close this issue for now. Feel free to open a new issue if you run into any further problems.
